perlmod/Fink/PkgVersion.pm

```diff
@@ -4105,22 +4105,37 @@ EOF
 
 	my $has_kernel_dep;
 	my $deps = $self->get_depends(0, 0); # get runtime dependencies
+	my $predeps = $self->pkglist('Pre-Depends');
 
-	foreach (@$deps) {
+	foreach (@$predeps) {
```

Comment:
Might be cleaner to unify this and the next block as:
`foreach (@$deps, @$predeps)`
For efficiency, could use a label and `last` to jump out of the *inner* loop, since we can stop as soon as we find a single match. IIRC, the kernel-dep is usually appended to the Depends, so it would be especially efficient if @$deps were scanned in reverse order. But cost of reversing a list vs the actual time saved, maybe not worth it to be clever:)